### PR TITLE
doc: update release-notes and bips.md for BIP111

### PR DESCRIPTION
**Pull request by laanwj**
Documents #6579

### CODE REVIEW

**Comment by luke-jr** [comment]
ACK

File: doc/release-notes.md [hunk 1]
```diff
@@ -58,6 +58,23 @@ specified. It used to be the case that `-X -noX` ends up, unintuitively, with X
...
+Support for the `NODE_BLOOM` service bit, as described in [`BIP
+111'](https://github.com/bitcoin/bips/blob/master/bip-0111.mediawiki), has been
+added to the P2P protocol code.
+
+BIP 111 defines a service bit to allow peers to advertise that they support
+bloom filters (such as used by SPV clients) explicitly. It also bumps the protocol
+version to allow peers to identify old nodes which allow bloom filtering of the
+connection despite lacking the new service bit.
+
+In this version, it only enforced for peer peers that send protocol versions
```
Comment (fanquake) [inline doc/release-notes.md]: s/it only/it is only
s/peers peers/peers

**Comment by dcousens** [comment]
ACK after fixes pointed out by @fanquake

File: doc/bips.md [hunk 2]
```diff
@@ -16,3 +16,4 @@ BIPs that are implemented by Bitcoin Core (up-to-date up to **v0.10.0**):
 * [`BIP 61`](https://github.com/bitcoin/bips/blob/master/bip-0061.mediawiki): The 'reject' protocol message (and the protocol version bump to 70002) was added in **v0.9.0** ([PR #3185](https://github.com/bitcoin/bitcoin/pull/3185)).
 * [`BIP 66`](https://github.com/bitcoin/bips/blob/master/bip-0066.mediawiki): The strict DER rules and associated version 3 blocks have been implemented since **v0.10.0** ([PR #5713](https://github.com/bitcoin/bitcoin/pull/5713)).
 * [`BIP 70`](https://github.com/bitcoin/bips/blob/master/bip-0070.mediawiki) [`71`](https://github.com/bitcoin/bips/blob/master/bip-0071.mediawiki) [`72`](https://github.com/bitcoin/bips/blob/master/bip-0072.mediawiki): Payment Protocol support has been available in Bitcoin Core GUI since **v0.9.0** ([PR #5216](https://github.com/bitcoin/bitcoin/pull/5216)).
+* [`BIP 111'](https://github.com/bitcoin/bips/blob/master/bip-0111.mediawiki): `NODE_BLOOM` service bit added, but only enforced for peer versions `>=70011` as of **v0.12.0**
```
Comment (MarcoFalke) [inline doc/bips.md]: Needs a link to the PR. See https://github.com/laanwj/bitcoin/pull/4.

**Comment by laanwj** [comment]
Cherry-picked @MarcoFalke's commit

**Pull request merged**